conn.go

```diff
@@ -184,12 +196,17 @@ func (c *Conn) rpc(opcode int32, w jute.RecordWriter, r jute.RecordReader) error
 	}
 }
 
-func (c *Conn) handleReads() {
+func (c *Conn) handleReadWrites() {
 	defer c.Close()
 	for {
 		if c.sessionCtx.Err() != nil {
 			return
 		}
+		wr := <-c.wchann
+		if err := WriteRecords(c.conn, wr.header, wr.writer); err != nil {
```

Comment:
this seems to work even if this call is made in a goroutine here. So the trick seems to be in not starting reads until we do a write?

Reply:
I think the issue is that writes are overwriting each other buffers before the data is sent on the wire.

Reply:
Which buffers?

Reply:
The TCP socket buffer

Reply:
According to [POSIX docs](https://pubs.opengroup.org/onlinepubs/9699919799/functions/V2_chap02.html#tag_15_09_07) these operations should be thread safe. Am I missing something?

Reply:
Also, one problem with this approach is that we block on reads. So waiting for a read to happen to be able to write data might not be the best thing to do.

Reply:
This is why I proposed using a resource pool.